external/netutils/netlib_netmon.c

```diff
@@ -0,0 +1,173 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#if defined(CONFIG_NET_NETMON) && CONFIG_NSOCKET_DESCRIPTORS > 0
+
+#include <sys/socket.h>
+#include <sys/ioctl.h>
+
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+
+#include <netinet/in.h>
+#include <net/if.h>
+
+#include <netutils/netlib.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#if defined(CONFIG_NET_IPv4)
+#define PF_INETX PF_INET
+#elif defined(CONFIG_NET_IPv6)
+#define PF_INETX PF_INET6
+#endif
+#define NUM_SOCKETS MEMP_NUM_NETCONN
```

Comment:
As I Know MEMP_NUM_NETCONN is constant value of lwip but now you making generic network monitor, aren't you?
If I am right, CONFIG_NSOCKET_DESCRIPTORS may correct.

Reply:
You are correct. Modification completed.